lnbits/core/tasks.py

```diff
@@ -188,3 +189,27 @@ async def purge_audit_data():
 
         # clean every hour
         await asyncio.sleep(60 * 60)
+
+
+async def collect_exchange_rates_data():
+    """
+    Collect exchange rates data. Used for monitoring only.
+    """
+    while settings.lnbits_running:
+        currency = settings.lnbits_default_accounting_currency or "USD"
+        max_history_size = settings.lnbits_exchange_history_size
+        sleep_time = settings.lnbits_exchange_history_refresh_interval_seconds
+
+        if sleep_time > 0:
+            try:
+                rates = await btc_rates(currency)
+                if rates:
+                    rates_values = [r[1] for r in rates]
+                    lnbits_rate = sum(rates_values) / len(rates_values)
+                    rates.append(("LNbits", lnbits_rate))
+                settings.append_exchange_rate_datapoint(dict(rates), max_history_size)
+            except Exception as ex:
+                logger.warning(ex)
+        else:
+            sleep_time = 60
```

Comment:
```suggestion
        sleep_time = settings.lnbits_exchange_history_refresh_interval_seconds
        sleep_time = sleep_time if sleep_time > 0 else 60

        try:
                rates = await btc_rates(currency)
                if rates:
                rates_values = [r[1] for r in rates]
                lnbits_rate = sum(rates_values) / len(rates_values)
                rates.append(("LNbits", lnbits_rate))
            settings.append_exchange_rate_datapoint(dict(rates), max_history_size)
        except Exception as ex:
            logger.warning(ex)
```

Reply:
if `sleep_time` is zero or negative it should not try to fetch the exchange rates